st2api/st2api/controllers/v1/pack_configs.py

```diff
@@ -0,0 +1,106 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pecan
+
+from st2common.constants.keyvalue import USER_SCOPE
+from st2common.services import packs as packs_service
+from st2common.models.api.base import jsexpose
+from st2api.controllers.resource import ResourceController
+from st2common.models.api.pack import ConfigAPI
+from st2common.models.api.pack import ConfigItemSetAPI
+from st2common.persistence.pack import Config
+from st2common.services.config import set_datastore_value_for_config_key
+from st2common.rbac.utils import get_user_db_from_request
+from st2common.rbac.types import PermissionType
+from st2common.rbac.decorators import request_user_has_permission
+from st2common.rbac.decorators import request_user_has_resource_db_permission
+
+__all__ = [
+    'PackConfigsController'
+]
+
+
+class PackConfigsController(ResourceController):
+    model = ConfigAPI
+    access = Config
+    supported_filters = {}
+
+    def __init__(self):
+        super(PackConfigsController, self).__init__()
+
+        # Note: This method is used to retrieve object for RBAC purposes and in
+        # this case, RBAC is checked on the parent PackDB object
+        self.get_one_db_method = packs_service.get_pack_by_ref
+
+    @request_user_has_permission(permission_type=PermissionType.PACK_LIST)
+    @jsexpose()
+    def get_all(self, **kwargs):
+        """
+        Retrieve configs for all the packs.
+
+        Handles requests:
+            GET /configs/
+        """
+        # TODO: Make sure secret values are masked
+
+        return super(PackConfigsController, self)._get_all(**kwargs)
+
+    @request_user_has_resource_db_permission(permission_type=PermissionType.PACK_VIEW)
+    @jsexpose(arg_types=[str])
+    def get_one(self, pack_ref):
+        """
+        Retrieve config for a particular pack.
+
+        Handles requests:
+            GET /configs/<pack_ref>
+        """
+        # TODO: Make sure secret values are masked
+        return self._get_one_by_pack_ref(pack_ref=pack_ref)
+
+    @jsexpose(arg_types=[str], body_cls=ConfigItemSetAPI)
+    def put(self, pack_ref, config_item_api):
```

Comment:
I am not sure if this is a good idea to do this without locks. In HA, this is obviously a problem. Just noting it so you capture in your TODO. It's probably easier to fix it now.